client/app/app.js

```diff
@@ -0,0 +1,51 @@
+import { h, Component } from 'preact'
+import { Router } from 'preact-router'
+import { I18n } from './plugins/preact-polyglot'
+
+import Sidebar from './components/sidebar'
+import Discovery from './components/discovery'
+import CategoryList from './components/category_list'
+import ConnectedList from './components/connected_list'
+import ItemDialog from './components/item_dialog'
+
+export default class App extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            konnectors: props.konnectors,
+            selectedItem: null
+        }
+        this.onItemSelect = this.onItemSelect.bind(this)
```

Comment:
I thought we could always use the decorator later, or that we could use a better router to handle that use case.